src/com/xilinx/rapidwright/edif/EDIFTools.java

```diff
@@ -906,9 +906,23 @@ public static EDIFNetlist createNewNetlist(String topName, boolean addINV){
 		EDIFDesign eDesign = new EDIFDesign(topName);
 		n.setDesign(eDesign);
 		eDesign.setTopCell(top);
-		
 		return n;
 	}
+
+	/**
+	 * Creates a new netlist from an existing EDIFCellInst in a netlist.  
+	 * @param cellInst The new top cell/top cell inst in the netlist.
+	 * @return The newly created netlist from the provided cell inst.
+	 */
+	public static EDIFNetlist createNewNetlist(EDIFCellInst cellInst){
+	    EDIFNetlist n = new EDIFNetlist(cellInst.getName());
+	    n.generateBuildComments();
+	    EDIFDesign eDesign = new EDIFDesign(cellInst.getName());
+	    n.setDesign(eDesign);
+	    eDesign.setTopCell(cellInst.getCellType());
+	    n.migrateCellAndSubCells(cellInst.getCellType());
```

Comment:
If I'm reading the code for this function correctly, this is a destructive operation with respect to the input netlist, right? If so (and I'm okay with it being so) would a more apt description be `moveToNewNetlist` and the doc to be updated accordingly?

Reply:
This is probably easy enough to make it non-destructive, I will see if I can change it so that it leaves the original netlist intact.

Reply:
I looked again at this code, I don't believe it is destructive to the source netlist.  Do you see anything that is?  

I can see sharing of `EDIFCell` objects between the original and new `Netlist` objects, but I don't see how the original netlist is changed.

Reply:
I may be quite wrong, but `migrateCellAndSubCells` gave me the impression that it was a "move" operation and not a "copy", and so when I dove into that method I saw this:
https://github.com/Xilinx/RapidWright/blob/3ebcff1a4816e092e0e38658ebb8945c8f886c01/src/com/xilinx/rapidwright/edif/EDIFNetlist.java#L436-L458
which seems to be adding the same cell reference into both netlists (is this wise?) and the cell's instances are then modified as more migration recursively happens.

Reply:
Yes, that seems right.  I wasn't thinking hard enough about it.  This method has always been used such that the source design was not needed for anything after, so it hasn't caused an issue.  I'll add some documentation that denotes this and we can create a non-destructive version whenever that is needed.